External/SPMmodified/xASL/xASL_im_FillNaNs.m

```diff
@@ -228,22 +228,27 @@
 % Create NaN mask for all 3D volumes together
 maskNaN = sum(isnan(IM), 4) > 0;
 
-% The voxel-size needs to be estimated
-if isempty(VoxelSize)
-	% For each volume, we calculate the difference of neighboring voxels
-	% We calculate the mean over nonNaN values, which gives the average update including the correct direction
-	diffIM = IM(2:end, :, :, 1) - IM(1:end-1, :, :, 1);
-	diffIM = diffIM(~maskNaN(2:end, :, :));
-	VoxelSize(1) = xASL_stat_MeanNan(diffIM);
+% We estimate the voxel-size and axis directions
+% For each volume, we calculate the difference of neighboring voxels
+% We calculate the mean over nonNaN values, which gives the average update including the correct direction
+diffIM = IM(2:end, :, :, 1) - IM(1:end-1, :, :, 1);
+diffIM = diffIM(~maskNaN(2:end, :, :));
+VoxelSizeEst(1) = xASL_stat_MeanNan(diffIM);
 
+diffIM = IM(:, 2:end, :, 2)-IM(:, 1:end-1, :, 2);
+diffIM = diffIM(~maskNaN(:, 2:end, :));
+VoxelSizeEst(2) = xASL_stat_MeanNan(diffIM);
 
-	diffIM = IM(:, 2:end, :, 2)-IM(:, 1:end-1, :, 2);
-	diffIM = diffIM(~maskNaN(:, 2:end, :));
-	VoxelSize(2) = xASL_stat_MeanNan(diffIM);
+diffIM = IM(:, :, 2:end, 3)-IM(:, :, 1:end-1, 3);
+diffIM = diffIM(~maskNaN(:, :, 2:end));
+VoxelSizeEst(3) = xASL_stat_MeanNan(diffIM);
 
-	diffIM = IM(:, :, 2:end, 3)-IM(:, :, 1:end-1, 3);
-	diffIM = diffIM(~maskNaN(:, :, 2:end));
-	VoxelSize(3) = xASL_stat_MeanNan(diffIM);
+if isempty(VoxelSize)
```

Comment:
No, we never use the VoxelSize. I would remove the VoxelSize option fully. Because we never used it in the pipeline, and to stay consistent, I would also not start using it.

Reply:
But the option is there, so why we have option that we are not using - that's very inconsistent.

But I can of course always automatically remove it.

Reply:
No the option is there only for METHOD 1, to calculate the smoothing kernel only. Not for METHOD 3, and I would not use it to stay consistent, to always do the same.

Reply:
OK